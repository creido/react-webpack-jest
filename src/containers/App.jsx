import React, { Component } from 'react';

import Header from '../components/Header';
import Main from '../components/Main';
import Footer from '../components/Footer';
import HelloWorld from '../components/HelloWorld';

const App = () => (
  <div>
    <Header/>
    <Main/>
    <Footer/>
    <HelloWorld/>
  </div>
);

export default App;
