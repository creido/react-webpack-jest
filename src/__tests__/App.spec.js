import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../containers/App';

describe('App', () => {
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('renders App component correctly', () => {
    const tree = renderer.create(
      <App />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
