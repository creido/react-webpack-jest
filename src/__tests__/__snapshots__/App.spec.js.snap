// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component correctly 1`] = `
<div>
  <header>
    blah header
  </header>
  <main>
    Main
  </main>
  <footer>
    Footer bits and pieces
  </footer>
  <div
    className="hello-world"
  >
    <h1>
      Hello World
    </h1>
  </div>
</div>
`;
